scripts/s3-support-bundles/s3_bundle_generate.sh

```diff
@@ -98,19 +98,20 @@ s3server_logdir=`cat $s3server_config | grep "S3_LOG_DIR:" | cut -f2 -d: | sed -
 authserver_logdir=`cat $authserver_config | grep "logFilePath=" | cut -f2 -d'=' | sed -e 's/^[ \t]*//' -e 's/#.*//' -e 's/^[ \t]*"\(.*\)"[ \t]*$/\1/'`
 backgrounddelete_logdir=`cat $backgrounddelete_config | grep "logger_directory:" | cut -f2 -d: | sed -e 's/^[ \t]*//' -e 's/#.*//' -e 's/^[ \t]*"\(.*\)"[ \t]*$/\1/'`
 
-# Compress each s3server core file present in /var/crash directory if available
+# Compress latest s3 core with <s3_core_files_max_count> files from /var/crash directory if available
 # these compressed core file will be available in /tmp/s3_support_bundle_<pid>/s3_core_files directory
 compress_core_files(){
-  core_filename_pattern="*/core-s3server.*"
-  for file in $s3_core_dir/*
+  core_filename_pattern="core-s3server.*"
+  # get recent modified core files from directory
+  core_files_list=$(ls -tr "$s3_core_dir" | grep "$core_filename_pattern" | tail -"$s3_core_files_max_count")
```

Comment:
I believe, "ls -tr" will return list with oldest file first(on the top of list) and latest at bottom. 
When you 'tail -x' on this list, your will get last x files. This will include the latest as well. But, we want to consider old files.
I think, you may need to use 'ls -t' OR 'head x'

Reply:
Probably, you may have to use 'ls -ltr' and than use head -n x to get the 'x' oldest files